src/github/mod.rs

```diff
@@ -0,0 +1,220 @@
+    } = repository;
+    let url =
+        format!("https://raw.githubusercontent.com/{repo_owner}/{repo_name}/{repo_branch}/{path}");
+    let response = reqwest::get(url).await?;
+    if response.status() == reqwest::StatusCode::OK {
+        let content = response.text().await?;
+        Ok(content)
+    } else {
+        Err(anyhow::anyhow!("Unable to fetch file content"))
+    }
+}
+
+const IGNORED_EXTENSIONS: &[&str] = &[
+    "bpg", "eps", "pcx", "ppm", "tga", "tiff", "wmf", "xpm", "svg", "ttf", "woff2", "fnt", "fon",
+    "otf", "pdf", "ps", "dot", "docx", "dotx", "xls", "xlsx", "xlt", "lock", "odt", "ott", "ods",
+    "ots", "dvi", "pcl", "mod", "jar", "pyc", "war", "ear", "bz2", "xz", "rpm", "coff", "obj",
+    "dll", "class", "log",
+];
+
+const IGNORED_DIRECTORIES: &[&str] = &[
+    "vendor",
+    "dist",
+    "build",
+    "target",
+    "bin",
+    "obj",
+    "node_modules",
+    "debug",
+];
+
+pub fn should_index(path: &str) -> bool {
+    !(IGNORED_EXTENSIONS.iter().any(|ext| path.ends_with(ext))
+        || IGNORED_DIRECTORIES.iter().any(|dir| path.contains(dir)))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_fetch_repo_files() {
+        let repository = Repository {
+            owner: "open-sauced".to_string(),
+            name: "ai".to_string(),
+            branch: "beta".to_string(),
+        };
+
+        let result = fetch_repo_files(&repository).await;
+
+        // Assert that the function returns a Result containing a vector of File
+        assert!(result.is_ok());
+        let files = result.unwrap();
+        assert!(files.len() > 0);
+    }
+
+    #[tokio::test]
+    async fn test_fetch_file_content() {
+        let repository = Repository {
+            owner: "open-sauced".to_string(),
+            name: "ai".to_string(),
+            branch: "beta".to_string(),
+        };
+        let path = "package.json";
+
+        let result = fetch_file_content(&repository, path).await;
+
+        // Assert that the function returns a Result containing the file content
+        assert!(result.is_ok());
+        let content = result.unwrap();
+        assert!(content.len() > 0);
+
+        let path = "Some_Invalid_File.example";
+
+        let result = fetch_file_content(&repository, path).await;
+
+        //Assert that the function returns Err for an invalid file path
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_should_index() {
+        // Test with ignored extensions
+        for ext in IGNORED_EXTENSIONS {
+            let path = format!("path/to/file.{}", ext);
+            assert!(!should_index(&path));
+        }
+
+        // Test with ignored directories
+        for dir in IGNORED_DIRECTORIES {
+            let path = format!("path/to/{}/file.txt", dir);
+            assert!(!should_index(&path));
+        }
+
+        // Test with valid path
+        let path = "path/to/file.tsx";
+        assert!(should_index(&path));
```

Comment:
Nit: `cargo clippy` wants to make this change:

```diff
-        assert!(should_index(&path));
+        assert!(should_index(path));
```

Not sure why it's not complaining about the other ones. Might be fine.

Reply:
The others were `String`s from `format!` that needed derefs.
https://github.com/open-sauced/repo-query/pull/1/commits/3ec893a487f6c806e750ec5e8e398d81024d9fd6